pkg/kubelet/kubelet.go

```diff
@@ -1657,7 +1657,7 @@ func (kl *Kubelet) deletePod(pod *v1.Pod) error {
 
 	// Runtime cache may not have been updated to with the pod, but it's okay
 	// because the periodic cleanup routine will attempt to delete again later.
-	runningPods, err := kl.runtimeCache.GetPods()
+	runningPods, err := kl.runtimeCache.GetPods(true)
```

Comment:
If this is the only place we are calling `GetPods(true)`, may be just call `getPodsWithTimestamp()` directly? is there a trade off?

Reply:
For now, yes.

`runtimeCache` is a `RuntimeCache` interface and `getPodsWithTimestamp()` isn't accessible and I didn't want to add another method to the interface that did mostly the same thing as `GetPods()` when I could just toggle the behavior with a boolean.

Reply:
how expensive is getting pods and missing the cache hit?

Reply:
The default housekeeping routine runs every 2s.  The cache expiry is also 2s.  So there is a call through the cache every 2s.  This only adds an additional call through the cache into the runtime on each pod delete.

It is expensive for the runtime, but we aren't doing it considerably more after this change.

In fact... know that I think about it... we probably aren't doing any more at all because forcing the update means that we the next call the housekeeping routine would do is cached.  We are just advancing a call that we would have done anyway.  Unless we are doing more that one pod delete every 2s.

Reply:
> Unless we are doing more than one pod delete every 2s.

"rapidly delete lots of pods" is the scenario I'm wondering about. 

> If a pod's container starts and the pod is deleted all within the 2s defaultCachePeriod of the runtime cache, then the grace periods of the containers may not be honored. This is because GetPods() can return a stale runningPod with no containers

is it possible to determine the last time we attempted a start of a container in the pod and pass that to GetPods as a sort of notOlderThan? Would that let us continue using the cached result in most cases, and only skip the cache when needed?

Reply:
maybe there is some way to link PLEG to the runtime cache... i'll look into it